### PR TITLE
New convo msg store

### PR DESCRIPTION
**Pull request by clague17**
Changes

- Changed parameters for both Conversations as well as Message Classes as per the design spec
- `Conversation(UUID id, UUID ownerId, Instant creationTime, String title, HashSet<UUID> members,
 Type type, Visibility visibility, String avatarImageURL, boolean isActive, ChronoUnit validTime, Instant deletionInstant, int totalPoints, String description, HashSet<UUID> haveVoted)` 
- Fixed all previous tests and errors in order to get a development server up and running. 
- `Message(UUID id, UUID conversaationId, UUID author, Pair content, Instant creationTime)`
- Where the content of a Message is stored as a tuple of values, <text, media>, or <String, BlobKey>.
- Created an Upload.java, Serve.java explicitly for working with the BlobStore aspect of loading content. 
What's next
- Being able to create a new conversation seamlessly with the new changes,
- Styling the new conversation page a bit better
- working on chat.jsp to display the actual message contents and thoroughly learning how to use BlobStore.

### CODE REVIEW

File: src/main/java/codeu/model/data/Message.java [hunk 1]
```diff
@@ -60,20 +104,75 @@ public UUID getAuthorId() {
...
+
+  public Pair getContent() {
+		// content is a tuple!
     return content;
   }
 
   /** Returns the creation time of this Message. */
   public Instant getCreationTime() {
-    return creation;
+    return creationTime;
```
[the remark this answers is not in the document]
Comment (clague17) [reply src/main/java/codeu/model/data/Message.java]: Yeah you are definitely right

[the remark this answers is not in the document]
Comment (clague17) [reply src/main/java/codeu/model/data/Message.java]: Yeah you are definitely right

File: src/main/java/codeu/controller/Upload.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		List<BlobKey> blobKeys = blobs.get("files");
+		//wanna have some descriptions to kinda keep track of this stuff!
+		//So one description could be like conversationAvatarImages, BlobKey
+        // BlobKeyCache bc = BlobKeyCache.getBlobKeyCache();
+
+        if (blobKeys == null){
+            System.out.println("blobkey is null");
+        }
+        else {
+        	// for(BlobKey blobkey:blobKeys){
```
[the remark this answers is not in the document]
Comment (clague17) [reply src/main/java/codeu/controller/Upload.java]: Haha I'm not quite ready to hash out the issues with blobkeys so thats why its commented

[the remark this answers is not in the document]
Comment (clague17) [reply src/main/java/codeu/controller/Upload.java]: Haha I'm not quite ready to hash out the issues with blobkeys so thats why its commented

File: src/main/java/codeu/controller/ConversationServlet.java [hunk 3]
```diff
@@ -116,63 +142,165 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
+			List<BlobKey> blobKeys = allBlobs.get("avatarImage/" + conversationTitle);
+			//by convention, the actual image should be stored in the last index of this blobKeys list.
+
+			BlobKey avatarBlobKey = blobKeys.get((blobKeys.size()-1));
+			ServingUrlOptions servingURLOptions = ServingUrlOptions.Builder.withBlobKey(avatarBlobKey);
+			avatarImageURL = imagesService.getServingUrl(servingURLOptions);
+			System.out.println("avatarImageURL");
+			System.out.println(avatarImageURL);
+
+			// response.sendRedirect("/serve?blob-key=" + blobKey.getKeyString());
```
[the remark this answers is not in the document]
Comment (clague17) [reply src/main/java/codeu/controller/ConversationServlet.java]: Yes should be commented, but will delete once I am done with testing and ensure this works

File: src/main/java/codeu/model/data/Message.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (clague17) [reply src/main/java/codeu/model/data/Message.java]: Yeah you are definitely right

File: pom.xml [hunk 4]
```diff
@@ -51,13 +51,25 @@
...
-    
+
     <dependency>
       <groupId>org.mindrot</groupId>
       <artifactId>jbcrypt</artifactId>
       <version>0.3m</version>
     </dependency>
 
+		<dependency>
+	    <groupId>org.javatuples</groupId>
```
[the remark this answers is not in the document]
Comment (clague17) [reply pom.xml]: I imported a library called javatuples in order to house the Message content. Basically the message content is now (text, blobkey) to make it easier to handle.

File: src/main/java/codeu/controller/ConversationServlet.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (cmondragon15117) [reply src/main/java/codeu/controller/ConversationServlet.java]: Okay

**Comment by cmondragon15117** [comment]
Looks like the branch has no merge conflicts, @clague17 if everything is working right, I'd merge and resolve possible issues

**Pull request merged**